### PR TITLE
Fix validation when accelerator is a string

### LINKED ISSUE #12698: Choosing an invalid accelerator does not raise an error
### To Reproduce

```py
trainer = Trainer(accelerator="invalid",)
print(type(trainer.accelerator))
```

### Expected behavior

Raises an error. 

The error is already implemented in the AcceleratorConnector:
https://github.com/PyTorchLightning/pytorch-lightning/blob/3f0f277d5bbf6f9c1f07e8f03e1d6ae048608d7a/pytorch_lightning/trainer/connectors/accelerator_connector.py#L490-L494

However, the code is never triggered. It was introduced in #12180 but not tested. 

### Environment

PL 1.7

cc @justusschock @kaushikb11 @awaelchli @ninginthecloud @akihironitta @rohitgr7

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #12698

Choosing `accelerator="cocofruit"` would silently convert it to `"cpu"`. 
Now, we will raise an error. See the added test.

### Does your PR introduce any breaking changes? If yes, please list them.

No.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/Lightning-AI/lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @justusschock @kaushikb11 @awaelchli @ninginthecloud @akihironitta @rohitgr7

### CODE REVIEW

File: src/pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 1]
```diff
@@ -496,12 +505,6 @@ def _set_parallel_devices_and_init_accelerator(self) -> None:
             self.accelerator: Accelerator = self._accelerator_flag
         else:
             assert self._accelerator_flag is not None
```
Comment (carmocca) [inline src/pytorch_lightning/trainer/connectors/accelerator_connector.py:507]: Is this assert still relevant?

File: tests/tests_pytorch/accelerators/test_accelerator_connector.py [hunk 2]
```diff
@@ -330,12 +326,6 @@ def test_accelerator_auto_with_devices_gpu():
     assert trainer.num_devices == 1
 
 
-def test_validate_accelerator_and_devices():
```
Comment (carmocca) [inline tests/tests_pytorch/accelerators/test_accelerator_connector.py:333]: Are you removing this because it's duplicated?

**Review by carmocca — APPROVED** [review]

File: src/pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 3]
```diff
@@ -275,9 +275,18 @@ def _check_config_and_set_final_flags(
...
-            if accelerator in self._accelerator_types or accelerator == "auto" or isinstance(accelerator, Accelerator):
-                self._accelerator_flag = accelerator
+        if (
+            accelerator is not None
+            and accelerator not in self._accelerator_types
+            and accelerator != "auto"
+            and not isinstance(accelerator, Accelerator)
+        ):
+            raise ValueError(
+                f"You selected an invalid accelerator name: `accelerator='{accelerator}'`."
```
Comment (rohitgr7) [inline src/pytorch_lightning/trainer/connectors/accelerator_connector.py:285]:
```suggestion
                f"You selected an invalid accelerator name: `accelerator={accelerator!r}`."
```
can be any primitive type.

**Review by rohitgr7 — APPROVED** [review]

File: tests/tests_pytorch/accelerators/test_accelerator_connector.py (hunk 2 above)
Comment (awaelchli) [reply tests/tests_pytorch/accelerators/test_accelerator_connector.py:333]: No but because now it would raise an error. This test seems like a forgotten artifact that silently passed because ddp_cpu was converted to just cpu. Now it is considered invalid and will raise.

File: src/pytorch_lightning/trainer/connectors/accelerator_connector.py (hunk 1 above)
Comment (awaelchli) [reply src/pytorch_lightning/trainer/connectors/accelerator_connector.py:507]: I think so because of the type check in mypy where AcceleratorRegistry.get() does not accept None as input.
